src/beamlime/handlers/data_reduction_handler.py

```diff
@@ -38,21 +37,19 @@ class ReductionHandlerFactory(
     def __init__(
         self,
         *,
-        instrument: str,
-        attrs_registry: Mapping[str, Mapping[str, Any]],
+        instrument_config: ModuleType,
         logger: logging.Logger | None = None,
         config: Config,
     ) -> None:
         self._logger = logger or logging.getLogger(__name__)
         self._config = config
-        self._instrument = instrument
-        self._attrs_registry = attrs_registry
-        instrument_config = get_config(instrument)
-        self._processors = instrument_config.make_stream_processors()
-        self._source_to_key = instrument_config.source_to_key
+        self._instrument_config = instrument_config
+        self._processors = self._instrument_config.make_stream_processors()
+        self._source_to_key = self._instrument_config.source_to_key
+        self._attrs_registry = self._instrument_config.f144_attribute_registry
```

Comment:
It is likely that I will turn `instrument_config` into a class in the future. I will try to figure out the details when trying to implement workflow parameter setup and restarts, which can likely be partially encapsulated in such a class.